File: exercise/ts/day14/src/children/dtos/child.ts
export class Child {
    Id!: string;
    FirstName?: string;
    MiddleName?: string;
    LastName?: string;
    BirthCity?: string;
    BirthDate!: string;
    Gender!: Gender;
    Address?: Address;
}

export enum Gender {
    Girl = "Girl",
    Boy = "Boy"
}

export class Address {
    Number!: string;
    Street!: string;
    City!: string;
    CountryId!: number;
}

File: exercise/C#/day14/Children/DTOs/Child.cs
namespace Children.DTOs
{
    public class Child
    {
        public Guid Id { get; set; }
        public string? FirstName { get; set; }
        public string? MiddleName { get; set; }
        public string? LastName { get; set; }
        public string? BirthCity { get; set; }
        public DateOnly BirthDate { get; set; }
        public Gender Gender { get; set; }
        public Address? Address { get; set; }
    }

    public enum Gender
    {
        Girl,
        Boy
    }

    public class Address
    {
        public string Number { get; set; }
        public string Street { get; set; }
        public string City { get; set; }
        public int CountryId { get; set; }
    }
}